gd1_helpers/membership/gd1_model.py

```diff
@@ -155,17 +294,78 @@ def extra_ln_prior(self, params):
 
         return lp
 
+    @classmethod
+    def stream_dens_update(cls, pawprint, data, knot_sep):
```

Comment:
Same question about this as above. If we don't like the static class implementation, maybe we should change the interface. Right now, there is some duplicated code defining the component variables inside and outside of your `update()` methods.